packages/test/test-end-to-end-tests/src/test/legacyChunking.spec.ts

```diff
@@ -0,0 +1,107 @@
+        let provider: ITestObjectProvider;
+        let oldMap: SharedMap;
+        let newMap: SharedMap;
+        beforeEach(() => {
+            provider = getTestObjectProvider();
+        });
+        afterEach(async () => provider.reset());
+
+        const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+            runtime.IFluidHandleContext.resolveHandle(request);
+        const mapId = "map";
+        const registry: ChannelFactoryRegistry = [
+            [mapId, SharedMap.getFactory()],
+        ];
+        const factory: TestFluidObjectFactory = new TestFluidObjectFactory(
+            registry,
+            "default",
+        );
+        const testContainerConfig: ITestContainerConfig = {
+            fluidDataObjectType: DataObjectFactoryType.Test,
+            registry,
+        };
+
+        const createOldContainer = async (): Promise<IContainer> => {
+            const oldContainerRuntimeFactoryWithDefaultDataStore =
+                getContainerRuntimeApi(versionWithChunking).ContainerRuntimeFactoryWithDefaultDataStore;
+            const oldRuntimeFactory =
+                new oldContainerRuntimeFactoryWithDefaultDataStore(
+                    factory,
+                    [
+                        [factory.type, Promise.resolve(factory)],
+                    ],
+                    undefined,
+                    [innerRequestHandler],
+                    {
+                        // Chunking did not work with FlushMode.TurnBased,
+                        // as it was breaking batching semantics. So we need
+                        // to force the container to flush the ops as soon as
+                        // they are produced.
+                        flushMode: FlushMode.Immediate,
+                        gcOptions: {
+                            gcAllowed: true,
+                        },
+                    },
+                );
+
+            return provider.createContainer(oldRuntimeFactory);
+        };
+
+        const setupContainers = async () => {
+            const oldContainer = await createOldContainer();
+            const oldDataObject = await requestFluidObject<ITestFluidObject>(oldContainer, "default");
+            oldMap = await oldDataObject.getSharedObject<SharedMap>(mapId);
+
+            const containerOnLatest = await provider.loadTestContainer(testContainerConfig);
+            const newDataObject = await requestFluidObject<ITestFluidObject>(containerOnLatest, "default");
+            newMap = await newDataObject.getSharedObject<SharedMap>(mapId);
+
+            await provider.ensureSynchronized();
+        };
+
+        const generateStringOfSize = (sizeInBytes: number): string => new Array(sizeInBytes + 1).join("0");
+
+        it("If an old container sends a large chunked op, a new container is able to process it successfully", async () => {
```

Comment:
How about old containers being able to process chunked ops from new container. Will that work?

Reply:
Yes it will! But not the type of chunked messages we plan on sending. We're planning on chunking compressed batches. So if the runtime doesn't support that, it's not going to work. But that's ok, as the feature is shipped dark. So the plan is to enable it in the future after we reach full saturation (or have enough confidence)

Reply:
I see - Not sure I understand what type of chunked messages will work for the scenario then.
Anyway, I was just wondering if we need tests where old containers can read chunked ops from new containers. For these new chunked op types, it seems we don't. But for the existing chunking, do we need it?

Reply:
The issue is that we (particularly me) will change the chunk reconstruction area (around `processRemoteChunkedMessage`) in order to enable the compressed chunked scenarios. I want to make sure we don't break the existing scenario..